src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1027,7 +1027,7 @@ public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
             if (typeof(T) == typeof(byte))
             {
                 int result = SpanHelpers.IndexOf(
-                    ref Unsafe.Add(ref array.GetRawSzArrayData(), startIndex),
+                    ref Unsafe.As<byte[]>(array).GetRawSzArrayData(startIndex),
```

Comment:
Not sure whether the helper takes index is worth it. If you are doing `ref` arithmetic, your code is going to be full of `Unsafe.*` anyway. One more or less does not seem to be a bit deal.